### ISSUE TITLE
ESLint error on prefer-default with export * in index.js

### ISSUE THREAD

**Issue by JSafaiyeh** [benmosher/eslint-plugin-import#897]
**Tell us about your environment**
* **ESLint Version:** ^3.19.0
* **Node Version:** v4.2.6
* **npm Version:** 5.1.0
**What parser (default, Babel-ESLint, etc.) are you using?** babel-eslint"@^7.2.3

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js



```

</details>

**What did you do? Please include the actual source code causing the issue.**

```js

{
    "extends": "airbnb",
    "parser": "babel-eslint",
    "globals": {
        "__DEV__": true
    },
    "plugins": [
        "react",
        "react-native",
        "jsx-a11y",
        "import"
    ],
    "rules": {
      ...
    }
}
```

**What did you expect to happen?**
I am using export * to export all components in a directory.
```js
import React, { Component } from 'react';

class MyComponent extends Component {
   ...
}
export { MyComponent };
```

**index.js**
```js
export * from './MyComponent';
```
`export { MyComponent }` results in an error in `prefer-default-export`

**Comment by ljharb** [comment]
That's the intended result. `MyComponent.js` should have `export default MyComponent`, and your index file should do `export { default as MyComponent } from './MyComponent';`

**Comment by JSafaiyeh** [comment]
That solved it thanks.
